src/library/type_context.cpp

```diff
@@ -4097,22 +4129,23 @@ expr type_context_old::preprocess_class(expr const & type,
     if (!u_replacements.empty())
         C = update_constant(C, to_list(C_levels));
     expr it2 = infer(C);
+    buffer<bool> is_out_param = class_out_param_deps(it);
+    int i = 0;
     for (expr & C_arg : C_args) {
         it2  = whnf(it2);
         if (!is_pi(it2))
             return type; /* failed */
         expr const & d = binding_domain(it2);
-        if (/* Case 1 */
-            is_class_out_param(d) ||
-            /* Case 2 */
-            depends_on_mvar(d, new_mvars)) {
+        lean_assert(is_out_param[i] == (is_class_out_param(d) || depends_on_mvar(d, new_mvars)));
```

Comment:
`lean_assert` is only run in debug builds (i.e., not on mathlib).  I'd just remove this check.